.agents/skills/benchmarking-perf/SKILL.md

```
@@ -26,48 +26,72 @@ description: "Run and analyze criterion benchmarks for performance-sensitive cha
+```rust
+pub async fn inject_concepts(&self, concepts: &[(String, HVec10240)]) -> Result<()> {
+    // Validate all inputs first → Batch insert → Single transaction for DB
+}
+```
+
+## Caching Pattern
+- Prefer cached values as `Arc<[T]>` for cheap `Arc::clone` hits
+- Avoid keying caches via temporary `Vec` materializations; hash fixed-size words/arrays directly
+- Cache hit rate target: >80% for repeated access patterns
 
 ## Adding a New Benchmark
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)